src/base/bittorrent/trackerentry.h

```diff
@@ -31,31 +31,38 @@
 
 #include <libtorrent/announce_entry.hpp>
 
+#include <QString>
 #include <QtGlobal>
 
-class QString;
+class QStringList;
+template <class T> class QVector;
 
 namespace BitTorrent
 {
+    class EndpointView;
+
+    enum class TrackerStatus
+    {
+        NotContacted = 1,
+        Working = 2,
+        Updating = 3,
+        NotWorking = 4,
+        WorkingWarning = 5
```

Comment:
I think this extra state is redundant.

Reply:
It is useful for TrackerEntry user decision whether it should care or not about the messages, without checking them. And also system message sometime show non-errors, so there is disambiguation. And before my changes tracker warning messages were always hidden, even tho there is `Warning` torrent status.

Reply:
It seems to be optional info so I really disagree to create any extra state based on it.
>It is useful for TrackerEntry user decision whether it should care or not about the messages, without checking them.

Doesn't you just hide some logic from TrackerEntry user so there is "checking for messages" always performed (to create this extra state) even if user don't need it?
>before my changes tracker warning messages were always hidden, even tho there is Warning torrent status.

You don't require extra state to fix this error.